components/inputs/input-date.js

```diff
@@ -211,12 +225,46 @@ class InputDate extends LocalizeStaticMixin(LitElement) {
 		super.updated(changedProperties);
 
 		changedProperties.forEach((oldVal, prop) => {
-			if (prop === 'value') {
-				this._formattedValue = this.value ? formatISODateInUserCalDescriptor(this.value) : '';
+			if (prop === '_dateTimeDescriptor' || prop === 'value') {
+				this._setFormattedValue();
 			}
 		});
 	}
 
+	_getContentWidth() {
+		const text = document.createElement('div');
+		document.body.appendChild(text);
+		text.style.fontFamily = 'inherit';
+		text.style.fontSize = '0.8rem';
+		text.style.fontWeight = '400';
+		text.style.letterSpacing = '0.02rem';
+		text.style.lineHeight = '1.4rem';
+		text.style.position = 'absolute';
+		text.style.width = 'auto';
+
+		// in some languages (e.g., fr) placeholderWidth is bigger, in others (e.g., zh) contentWidth is bigger
+		text.textContent = (this._dateTimeDescriptor.formats.dateFormats.short).toUpperCase();
+		const placeholderWidth = text.getBoundingClientRect().width;
```

Comment:
Each `getBoundingClientRect()` call is going to force a paint (obviously). I wonder if it would be nicer performance wise to nest these inside `requestAnimationFrame` calls... @dbatiste any opinions here? This is what fastdom does under the covers.

The code would become lest clean and more nest-y though.

Reply:
Yeah it would be nice to avoid this.  This does seem kinda hacky.  Perhaps we could include these things in the template, with visibility: hidden to force the component to be at least that big without needing to create it elsewhere, measure it, and then update with the measured width... ?

Reply:
Not a terrible idea... we'd need to hide them from AT as well somehow.

Reply:
Thanks for the suggestion - I'll give this a go